webapp/graphite/render/datalib.py

```diff
@@ -167,7 +185,7 @@ def _fetchData(pathExpr, startTime, endTime, now, requestContext, seriesList):
       else:
         known_nones = len([val for val in known if val is None])
 
-      if known_nones > candidate_nones:
+      if known_nones > candidate_nones and len(series):
```

Comment:
isn't this change going to mean that an empty result will always overwrite the `seriesList` entry down in the `else` clause?